cloud/src/aws/mod.rs

```diff
@@ -0,0 +1,470 @@
+pub mod aws_tonbo_svc;
+pub mod flight_svc;
+
+use std::{
+    env,
+    marker::Send,
+    net::{Ipv4Addr, SocketAddr},
+    ops::Bound,
+    pin::Pin,
+    sync::Arc,
+};
+
+use arrow_flight::flight_service_server::FlightServiceServer;
+use async_stream::try_stream;
+use async_trait::async_trait;
+use fusio::{
+    path::Path as TonboPath,
+    remotes::aws::{fs::AmazonS3Builder, AwsCredential},
+    DynFs,
+};
+use futures_core::Stream;
+use futures_util::StreamExt;
+use tokio::fs::create_dir_all;
+use tonbo::{
+    arrow::array::RecordBatch,
+    executor::tokio::TokioExecutor,
+    parquet::errors::ParquetError,
+    record::{
+        dynamic::Value, util::records_to_record_batch, ArrowArrays, ArrowArraysBuilder, DynRecord,
+        DynRecordImmutableArrays, Record, RecordRef,
+    },
+    transaction::Transaction,
+    DbOption, Entry, DB,
+};
+use tonic::transport::Server;
+
+use crate::{
+    aws::{aws_tonbo_svc::AWSTonboSvc, flight_svc::TonboFlightSvc},
+    error::CloudError,
+    gen::grpc::{
+        self,
+        aws_tonbo_server::{AwsTonbo as AwsTonboRPC, AwsTonboServer},
+    },
+    ScanRequest, TonboCloud,
+};
+
+pub const DEFAULT_PORT: u32 = 8080;
+
+/// Every table has its own tonbo cloud instance.
+#[allow(dead_code)]
+pub struct AWSTonbo {
+    // TODO: Add Tonbo DB instance
+    // Name of the Tonbo cloud instance
+    name: String,
+    // Local Tonbo instance
+    tonbo: DB<DynRecord, TokioExecutor>,
+    // Remote file system
+    s3_fs: Arc<dyn DynFs>,
```

Comment:
We need to determine which I/O offload to S3, it might not the duty of this crate, we have to make the strategy clear in Tonbo